src/files-breadcrumbs.tsx

```diff
@@ -51,6 +59,138 @@ function useHostname() {
     return hostname;
 }
 
+function BookmarkButton({ path }: { path: string[] }) {
+    const [isOpen, setIsOpen] = React.useState(false);
+    const [user, setUser] = React.useState<cockpit.UserInfo | null>(null);
+    const [bookmarks, setBookmarks] = React.useState<string[]>([]);
+    const [bookmarkHandle, setBookmarkHandle] = React.useState<cockpit.FileHandle<string> | null>(null);
+
+    const { cwdInfo } = useFilesContext();
+
+    const currentPath = "file://" + (path.join("/") || "/");
+    const defaultBookmarks = [];
+    if (user?.home)
+        defaultBookmarks.push({ name: _("Home"), loc: user?.home }); // TODO: add trash
+
+    useInit(async () => {
+        const user_info = await cockpit.user();
+        setUser(user_info);
+
+        const handle = cockpit.file(`${user_info.home}/.config/gtk-3.0/bookmarks`);
+        setBookmarkHandle(handle);
+
+        handle.watch((content) => {
+            if (content !== null) {
+                setBookmarks(content.split("\n").filter(line => line.startsWith("file://"))
+                        .map(line => {
+                            // Drop everything after the space, we don't show renames
+                            line = line.replace(/\s.*/, '');
+
+                            // Nautilus decodes urls as paths can contain spaces
+                            return decodeURI(line);
+                        }));
+            } else {
+                setBookmarks([]);
+            }
```

Comment:
As when saving I feel like instead of doing this as a special case we could just treat a missing file (`null`) as equivalent to an empty file (`''`), so: `(content || '').split(...)...`

Reply:
Could, or be explicit.

Reply:
It's a matter of taste.

My logic is that since the "file is empty" and "file does not exist" cases are handled in exactly the same way, they should be handled on the same path, instead of introducing a special case.

But definitely a taste thing, and I'm not about to insist on this.